### PR TITLE
Assorted improvements to the whole site

### PR DESCRIPTION
**Pull request by mikairi**
Improved the login experience, so that it automatically redirects when unauthenticated. Also rewrote the views to use some more Bootstrap so the sections are divided better.

The fullcalendar library is refactored into Bower. Make sure to run `bower install` from the SE344 project folder after you merge.

Really, just read the commit logs. :point_right:

### CODE REVIEW

**Comment by rayrobdod** [comment]
The entire point behind async is voided when you use await on the method call. Is there anywhere where things are being done asynchronously, or are we tagging methods with async solely because async is fun to say?

**Comment by mikairi** [comment]
If you don't use async in your service method, your controller method that calls the service will also be synchronous. Controller methods called by the framework can and should be made async if it deals with blocking operations like I/O.

Think about when multiple users try to hit the same page -> same controller method at the same time. Synchronous services would result in synchronous controllers like this:

``` csharp
public IActionResult DisplayStocksInfo() {
    ViewData["Price"]  = service.CurrentPrice(id);
    return View();
}
```

So each user has to wait for the previous user's Yahoo API query to return and processed before their own request is processed. That's the nature of synchronous code.

Now if your service method is async:

``` csharp
public async Task<IActionResult> DisplayStocksInfo() {
    ViewData["Price"] = await service.CurrentPrice(id);
    return View();
}
```

Although there isn't any work being done between the async operation and the return, the method itself is async and can called multiple times asynchronously by the framework to serve concurrent requests.

TL;DR: It is not necessarily the `GetResponse` method that inherently needs to be turned into `GetResponseAsync`, but it is a means to make the controller method that will call that service non-blocking. I do agree that it's not important for a project of our scale, but it is good practice regardless.

**Comment by rayrobdod** [comment]
So, the answer to my question is "Stuff can get called asynchronously inside the framework magic."

**Comment by mikairi** [comment]
Yes.

**Comment by rayrobdod** [comment]
If this is what's on the production server, then I approve.
